$OpenBSD$

1. PREFIX has to be /usr/local for OpenBSD
2. Do not gzip manpages on OpenBSD
3. BSD tar does not support "--exclude" option.
   See https://github.com/MLton/mlton/issues/146 for more detail

--- Makefile.orig	Mon Jul 15 19:59:09 2013
+++ Makefile	Thu May 26 13:01:56 2016
@@ -284,7 +284,7 @@ check:
 # puts them.
 DESTDIR := $(CURDIR)/install
 PREFIX := /usr
-ifeq ($(findstring $(TARGET_OS), darwin freebsd solaris), $(TARGET_OS))
+ifeq ($(findstring $(TARGET_OS), darwin freebsd openbsd solaris), $(TARGET_OS))
 PREFIX := /usr/local
 endif
 ifeq ($(TARGET_OS), mingw)
@@ -303,7 +303,7 @@ endif
 TEXM := $(TDOC)/examples
 
 GZIP_MAN := true
-ifeq ($(TARGET_OS), solaris)
+ifeq ($(findstring $(TARGET_OS), openbsd solaris), $(TARGET_OS))
 GZIP_MAN := false
 endif
 
@@ -399,7 +399,7 @@ move-docs:	install-docs install-no-docs
 
 .PHONY: release
 release: version
-	tar cvzf ../mlton-$(VERSION).src.tgz \
+	gtar cvzf ../mlton-$(VERSION).src.tgz \
 		--exclude .git --exclude package \
 		--transform "s@^@mlton-$(VERSION)/@S" \
 		*
